### PR TITLE
zstd: add static option

### PR DESCRIPTION
**Pull request by tobim**
CMake is used for static builds because the Makefile does not
offer a way to disable the shared library.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @orivej

### CODE REVIEW

**Comment by c0bw3b** [comment]
Isn't it possible to build both shared & static in one common multi-outputs derivation (with CMake) ; and then install static lib in "static" output ?

**Comment by tobim** [comment]
The regular Makefile based build already builds `libzstd` as shared object and static library in the same tree. A separate `static` output would be very unusual, almost no other package does that.

The purpose of this PR is mostly to get `zstd` to build in the `pkgsStatic` set, where position independent code is disabled and shared libraries don't link.

File: pkgs/tools/compression/zstd/default.nix [hunk 1]
```diff
@@ -59,4 +60,15 @@ stdenv.mkDerivation rec {
     platforms = platforms.unix;
     maintainers = with maintainers; [ orivej ];
   };
-}
+} // stdenv.lib.optionalAttrs static {
+  nativeBuildInputs = [ cmake ];
```
Comment (matthewbauer) [inline pkgs/tools/compression/zstd/default.nix]: It would be good if we can use cmake on non-static as well.

Comment (tobim) [reply pkgs/tools/compression/zstd/default.nix]: Done in the second commit.

**Comment by tobim** [comment]
superseded by #75798.

**Pull request closed without merging**